x-pack/plugins/transform/public/app/sections/transform_management/components/edit_transform_flyout/use_edit_transform_flyout.ts

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isEqual } from 'lodash';
+import { useReducer } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { TransformPivotConfig } from '../../../../common';
+
+const stringNotValidErrorMessage = i18n.translate(
+  'xpack.transform.transformList.editFlyoutFormStringNotValidErrorMessage',
+  {
+    defaultMessage: 'Value needs to be of type string.',
+  }
+);
+
+type Validator = (arg: any) => string[];
+
+// The way the current form is set up,
+// this validator is just a sanity check,
+// it should never trigger an error.
+const stringValidator: Validator = arg =>
+  typeof arg === 'string' ? [] : [stringNotValidErrorMessage];
+
+const frequencyNotValidErrorMessage = i18n.translate(
+  'xpack.transform.transformList.editFlyoutFormFrequencyNotValidErrorMessage',
+  {
+    defaultMessage: 'The frequency value is not valid.',
+  }
+);
+
+// Only allow frequencies in the form of 1s/1h etc.
+// Note this doesn't do a check against 0s yet.
+const frequencyValidator: Validator = arg => {
+  return /^([0-9]+[d|h|m|s|])$/.test(arg) ? [] : [frequencyNotValidErrorMessage];
```

Comment:
Could this be enhanced to check it is a valid interval between 1s and 1h?

Reply:
Improved checks and added unit tests in ef0daab.